api/openstorage.yaml

```diff
@@ -901,48 +923,107 @@ paths:
                     "background_processing": true
                     }]
                   }
-    put:
-      summary: Change volume state
-      description: |
-        Attach, detach, format, mount and unmount operations
-      parameters:
-        - name: uuid
-          in: path
-          description: Volume UUID
-          required: true
-          type: string
-          format: uuid
-        - name: state
-          in: body
-          description: Desired Volume State
-          required: true
-          schema:
-            $ref: '#/definitions/VolumeStateParams'
-      tags:
-        - Volumes
-      responses:
-        '200':
-          description: Volume reflecting current state
-          schema:
-            $ref: '#/definitions/Volume'
-    delete:
-      summary: Delete Volume
-      description: |
-        Delete Volume
-      parameters:
-        - name: uuid
-          in: path
-          description: Volume UUID
-          required: true
-          type: string
-          format: uuid
-      tags:
-        - Volumes
-      responses:
-        '200':
-          description: Volume Deleted
-          schema:
-            $ref: '#/definitions/UUID'
+      put:
+        summary: Change volume state
+        description: |
+          Attach, detach, format, mount and unmount operations
+        parameters:
+          - name: uuid
+            in: path
+            description: |
+             Volume UUID as path parameter example:
+             PUT /v1/osd-volumes/myvolume
+            required: true
+            type: string
+            format: uuid
+          - name: state
+            in: body
+            description: |
+              Desired Volume State attach example:
+               ``` json
+                   {
+                     "action": {
+                        "attach: 2"
```

Comment:
In the api.pb.go

// VolumeStateAction specifies desired actions.
type VolumeStateAction struct {
	// Attach or Detach volume
	Attach VolumeActionParam `protobuf:"varint,1,opt,name=attach,enum=openstorage.api.VolumeActionParam" json:"attach,omitempty"`
	// Mount or unmount volume
	Mount     VolumeActionParam `protobuf:"varint,2,opt,name=mount,enum=openstorage.api.VolumeActionParam" json:"mount,omitempty"`
	MountPath string            `protobuf:"bytes,3,opt,name=mount_path,json=mountPath" json:"mount_path,omitempty"`
	// Device path returned in attach
	DevicePath string `protobuf:"bytes,4,opt,name=device_path,json=devicePath" json:"device_path,omitempty"`
}